main.tf

```diff
@@ -92,3 +100,57 @@ resource "kubernetes_namespace" "namespace" {
     labels = each.value.labels
   }
 }
+
+resource "kubernetes_network_policy" "isolated" {
+  for_each = {
+    for p in local.drupal_network_policy_list : p.namespace => p.project_name if p.network_policy == "isolated"
+  }
+
+  metadata {
+    name      = "network-policy-isolated-${each.value}"
+    namespace = each.key
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+  }
+
+  depends_on = [
+    kubernetes_namespace.namespace
+  ]
+}
+
+resource "kubernetes_network_policy" "restricted" {
+  for_each = {
+    for p in local.drupal_network_policy_list : p.namespace => p.project_name if p.network_policy == "restricted"
+  }
+
+  metadata {
+    name      = "network-policy-restricted-${each.value}"
+    namespace = each.key
+  }
+
+  spec {
+    pod_selector {}
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      from {}
+    }
+  }
+
+  depends_on = [
+    kubernetes_namespace.namespace
+  ]
```

Comment:
I would prefer the `kubernetes.namespace[<key>].<namespace name argument>` in the `metadata.namespace` field to avoid explicit dependency.

What do you think?